### PR TITLE
fixed bug where app grain securable permissions were not resolved cor…

### PR DESCRIPTION
**Pull request by bsmithvt**
…rectly from access control UI

### CODE REVIEW

File: Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts [hunk 1]
```diff
@@ -8,26 +8,27 @@ import { Observable, of } from 'rxjs';
...
     const currentTimeStamp = new Date().getTime();
-    if (this.lastUpdateTimestamp === 0 || (Math.abs(currentTimeStamp - this.lastUpdateTimestamp) / 60000) > 2) {
-      this.lastUpdateTimestamp = currentTimeStamp;
-      return this.authUserService.getCurrentUserPermissions().pipe(tap(userPermissionResponse => {
-        this.permissions = userPermissionResponse.permissions;
+    if (!this.lastUpdateTimestampMap.has(securableItem)
+        || this.lastUpdateTimestampMap.get(securableItem) === 0
+        || (Math.abs(currentTimeStamp - this.lastUpdateTimestampMap.get(securableItem)) / 60000) > 2) {
+          this.lastUpdateTimestampMap.set(securableItem, currentTimeStamp);
+          return this.authUserService.getCurrentUserPermissions(securableItem).pipe(tap(userPermissionResponse => {
```
Comment (ctediz) [inline Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts]: overall, getting user permissions for a specified securable item/caching per securable right?

What was the issue before? Was access being restricted based off incorrect securables?

Comment (bsmithvt) [reply Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts]: @ctediz, correct. Rusty discovered an issue where app-grain securables could not be managed even if the user was tied to a role that had the `manageauthorization` permission for the selected app-grain securable. The `/user/permissions` route will set the securable item to the top-level securable owned by the `ClientId` in the access token if it isn't specified in the request, which in this case is `fabric-access-control` (see `SetDefaultRequest` in `UsersModule`). This prevents any permissions for the selected securable from being returned in the response.

File: Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts [hunk 2]
```diff
@@ -8,26 +8,27 @@ import { Observable, of } from 'rxjs';
...
   }
 
-  getPermissions(): Observable<string[]> {
+  getPermissions(securableItem: string): Observable<string[]> {
     const currentTimeStamp = new Date().getTime();
-    if (this.lastUpdateTimestamp === 0 || (Math.abs(currentTimeStamp - this.lastUpdateTimestamp) / 60000) > 2) {
-      this.lastUpdateTimestamp = currentTimeStamp;
-      return this.authUserService.getCurrentUserPermissions().pipe(tap(userPermissionResponse => {
-        this.permissions = userPermissionResponse.permissions;
+    if (!this.lastUpdateTimestampMap.has(securableItem)
```
Comment (hc-jordenlowe) [inline Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts]: This is starting to get complex.  Maybe throwing this into a new function would be good.  Perhaps this logic is an object?

Just a thought

Comment (bsmithvt) [reply Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts]: Good call - I just incorporated that suggestion.

**Pull request merged**